### PR TITLE
Allow creating TreeNode with ID different from the label

### PR DESCRIPTION
**Pull request by ptxmac**

### CODE REVIEW

File: wrapper/Widgets.cpp [hunk 1]
```diff
@@ -198,6 +198,12 @@ IggBool iggCollapsingHeader(const char *label, int flags)
    return ImGui::CollapsingHeader(label, flags) ? 1 : 0;
 }
 
+
+IggBool iggTreeNodeID(char const *id, char const *label, int flags)
+{
+   return ImGui::TreeNodeEx(id, flags, label) ? 1 : 0;
```
Comment (ptxmac) [inline wrapper/Widgets.cpp:204]: small caveat - the `label` is actually a format string in this call. Unfortunately, there's no version of `TreeNode*` that allows a specific ID and a plain label string

File: Widgets.go [hunk 2]
```diff
@@ -871,6 +871,15 @@ const (
 	TreeNodeFlagsCollapsingHeader = TreeNodeFlagsFramed | TreeNodeFlagsNoTreePushOnOpen | TreeNodeFlagsNoAutoOpenOnLog
 )
 
+// TreeNodeIDV is similar to TreeNodeV but sets a ID different from the label
```
Comment (ptxmac) [inline Widgets.go:874]:
```suggestion
// TreeNodeIDV is similar to TreeNodeV but sets a ID different from the label.
```

File: Widgets.go [hunk 3]
```diff
@@ -871,6 +871,15 @@ const (
 	TreeNodeFlagsCollapsingHeader = TreeNodeFlagsFramed | TreeNodeFlagsNoTreePushOnOpen | TreeNodeFlagsNoAutoOpenOnLog
 )
 
+// TreeNodeIDV is similar to TreeNodeV but sets a ID different from the label.
+func TreeNodeIDV(id string, label string, flags int) bool {
```
Comment (ptxmac) [inline Widgets.go:875]: I'm not super happy about the name, but it's already a bit weird. I.e. `TreeNodeV` actually calls the `TreeNodeEx` cpp function..

**Comment by ptxmac** [comment]
The test failed because of a timeout it seems

**Comment by dertseha** [comment]
Hello there and thank you for your contribution!
I'm a little occupied, I'll try to get to this issue soon. I have some thoughts on how to address the whole thing, yet haven't had time to fully cross-check and think it through.

**Comment by essial** [comment]
I'm confused, why would you not just do "MyTreeNode##WhateverElseForUniqueness"? Imgui only shows things to the left of ## allowing you to have multiple elements with the same 'text' but different IDs.

**Comment by ptxmac** [comment]
> I'm confused, why would you not just do "MyTreeNode##WhateverElseForUniqueness"? Imgui only shows things to the left of ## allowing you to have multiple elements with the same 'text' but different IDs.

Unfortunately, that doesn't work - it will only show the left side, but the whole text is used as the ID. If I change part of the text, the internal state (like whether or not the node is collapsed) will be reset.

**Comment by the-goodies** [comment]
I think this should solve your issue (taken from imgui [ FAQ](https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-why-is-my-widget-not-reacting-when-i-click-on-it)):

- Occasionally/rarely you might want change a label while preserving a constant ID. This allows you to animate labels. For example you may want to include varying information in a window title bar, but windows are uniquely identified by their ID. Use "###" to pass a label that isn't part of ID:

```
Button("Hello###ID");  // Label = "Hello",  ID = hash of (..., "###ID")
Button("World###ID");  // Label = "World",  ID = hash of (..., "###ID")  // Same ID, different label

sprintf(buf, "My game (%f FPS)###MyGame", fps);
Begin(buf);            // Variable title,   ID = hash of "MyGame"
```

**Comment by ptxmac** [comment]
> I think this should solve your issue (taken from imgui [ FAQ](https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-why-is-my-widget-not-reacting-when-i-click-on-it)):
> 
> * Occasionally/rarely you might want change a label while preserving a constant ID. This allows you to animate labels. For example you may want to include varying information in a window title bar, but windows are uniquely identified by their ID. Use "###" to pass a label that isn't part of ID:
> 
> ```
> Button("Hello###ID");  // Label = "Hello",  ID = hash of (..., "###ID")
> Button("World###ID");  // Label = "World",  ID = hash of (..., "###ID")  // Same ID, different label
> 
> sprintf(buf, "My game (%f FPS)###MyGame", fps);
> Begin(buf);            // Variable title,   ID = hash of "MyGame"
> ```

Thank you! I missed the part about `##` being different from `###`!
Using 3 `#`'s it does work using:
```
	nodeOpen := imgui.TreeNodeV(fmt.Sprintf("%s###%s", n.Name, n.ID), flags)
```

This solves my problem, so feel free to discard this PR unless there's another case where it makes sense

**Comment by dertseha** [comment]
Thank you all for joining in on this to help resolve the issue.
It would have been tricky (read: requiring a new major version) to support this extra variant with keeping the same name; Good that this was cleared.

Side-note: In case a format string parameter is in effect, to avoid escaping the text, instead of `printf(user_text)` it's better to use `printf("%s", user_text)`.
Discarding with confirmation.

**Pull request closed without merging**